### PR TITLE
DROOLS-5653 DMN fix stronglytyped deserialization for collection of temporals

### PR DESCRIPTION
**Pull request by tarilabs**
**JIRA**: https://issues.redhat.com/browse/DROOLS-5653

**referenced Pull Requests**: none

<pre>
How to retest a PR or trigger a specific build:

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</pre>

### CODE REVIEW

**Review by tkobayas — APPROVED** [review]

File: kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/typesafe/DMNDeclaredField.java [hunk 1]
```diff
@@ -102,8 +102,14 @@ public String getInitExpr() {
     public List<AnnotationDefinition> getFieldAnnotations() {
         List<AnnotationDefinition> annotations = new ArrayList<>();
         if (codeGenConfig.isWithJacksonAnnotation()) {
-            Optional<Class<?>> as = index.getJacksonDeserializeAs(fieldDMNType);
-            as.ifPresent(asClass -> annotations.add(new SimpleAnnotationDefinition("com.fasterxml.jackson.databind.annotation.JsonDeserialize").addValue("as", asClass.getCanonicalName() + ".class")));
+            boolean isCollection = fieldDMNType.isCollection();
+            DMNType verifyIfTemporal = fieldDMNType;
```
Comment (lucamolteni) [inline kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/typesafe/DMNDeclaredField.java]: "verify"? perhaps you meant "verifiedTemporal"? 

I'd avoid using the reassignment and I'd rather use two variables

**Review by lucamolteni — APPROVED** [review]

**Review by hellowdan — APPROVED** [review]

**Comment by tarilabs** [comment]
Jenkins retest this

**Comment by tarilabs** [comment]
Jenkins retest this

**Comment by tarilabs** [comment]
Jenkins run cdb

**Review by mariofusco — APPROVED** [review]

**Pull request merged**